File: Test/AdventureWorksFunctionalModel/Mapping/EmployeeMap.cs
using System.Data.Entity.ModelConfiguration;
using AW.Types;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace AW.Mapping {
    public class EmployeeMap : EntityTypeConfiguration<Employee> {
        public EmployeeMap() {
            // Primary Key
            HasKey(t => t.BusinessEntityID);

            // Properties
            Property(t => t.NationalIDNumber)
                .IsRequired()
                .HasMaxLength(15);

            Property(t => t.LoginID)
                .IsRequired()
                .HasMaxLength(256);

            Property(t => t.JobTitle)
                .IsRequired()
                .HasMaxLength(50);

            Property(t => t.MaritalStatus)
                .IsRequired()
                .IsFixedLength()
                .HasMaxLength(1);

            Property(t => t.Gender)
                .IsRequired()
                .IsFixedLength()
                .HasMaxLength(1);

            // Table & Column Mappings
            ToTable("Employee", "HumanResources");
            Property(t => t.BusinessEntityID).HasColumnName("BusinessEntityID");
            Property(t => t.NationalIDNumber).HasColumnName("NationalIDNumber");
            Property(t => t.LoginID).HasColumnName("LoginID");
            Property(t => t.JobTitle).HasColumnName("JobTitle");
            Property(t => t.DateOfBirth).HasColumnName("BirthDate");
            Property(t => t.MaritalStatus).HasColumnName("MaritalStatus");
            Property(t => t.Gender).HasColumnName("Gender");
            Property(t => t.HireDate).HasColumnName("HireDate");
            Property(t => t.Salaried).HasColumnName("SalariedFlag");
            Property(t => t.VacationHours).HasColumnName("VacationHours");
            Property(t => t.SickLeaveHours).HasColumnName("SickLeaveHours");
            Property(t => t.Current).HasColumnName("CurrentFlag");
            Property(t => t.rowguid).HasColumnName("rowguid");
            Property(t => t.ModifiedDate).HasColumnName("ModifiedDate"); //.IsConcurrencyToken();

            // Relationships
            Ignore(t => t.Manager);
            HasOptional(t => t.SalesPerson).WithRequired(t => t.EmployeeDetails);
            HasRequired(t => t.PersonDetails).WithOptional(t => t.Employee);
        }
    }

    public static partial class Mapper {
        public static void Map(this EntityTypeBuilder<Employee> builder) {
            builder.HasKey(t => t.BusinessEntityID);

            // Properties
            builder.Property(t => t.NationalIDNumber)
                   .IsRequired()
                   .HasMaxLength(15);

            builder.Property(t => t.LoginID)
                   .IsRequired()
                   .HasMaxLength(256);

            builder.Property(t => t.JobTitle)
                   .IsRequired()
                   .HasMaxLength(50);

            builder.Property(t => t.MaritalStatus)
                   .IsRequired()
                   .IsFixedLength()
                   .HasMaxLength(1);

            builder.Property(t => t.Gender)
                   .IsRequired()
                   .IsFixedLength()
                   .HasMaxLength(1);

            // Table & Column Mappings
            builder.ToTable("Employee", "HumanResources");
            builder.Property(t => t.BusinessEntityID).HasColumnName("BusinessEntityID");
            builder.Property(t => t.NationalIDNumber).HasColumnName("NationalIDNumber");
            builder.Property(t => t.LoginID).HasColumnName("LoginID");
            builder.Property(t => t.JobTitle).HasColumnName("JobTitle");
            builder.Property(t => t.DateOfBirth).HasColumnName("BirthDate");
            builder.Property(t => t.MaritalStatus).HasColumnName("MaritalStatus");
            builder.Property(t => t.Gender).HasColumnName("Gender");
            builder.Property(t => t.HireDate).HasColumnName("HireDate");
            builder.Property(t => t.Salaried).HasColumnName("SalariedFlag");
            builder.Property(t => t.VacationHours).HasColumnName("VacationHours");
            builder.Property(t => t.SickLeaveHours).HasColumnName("SickLeaveHours");
            builder.Property(t => t.Current).HasColumnName("CurrentFlag");
            builder.Property(t => t.rowguid).HasColumnName("rowguid");
            builder.Property(t => t.ModifiedDate).HasColumnName("ModifiedDate"); //.IsConcurrencyToken();

            // Relationships
            builder.Ignore(t => t.Manager);
            builder.HasOne(t => t.PersonDetails).WithOne(t => t.Employee);
        }
    }
}

File: Test/AdventureWorksLegacyModelVB/Mapping/EmployeeMap.vb
Imports System.Data.Entity.ModelConfiguration

Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
#Disable Warning BC8603
#Disable Warning BC8602

Namespace AW.Mapping
	Public Class EmployeeMap
		Inherits EntityTypeConfiguration(Of Employee)

		Public Sub New()
			' Primary Key
			HasKey(Function(t) t.BusinessEntityID)

			' Properties
			[Property](Function(t) t.NationalIDNumber).IsRequired().HasMaxLength(15)

			[Property](Function(t) t.LoginID).IsRequired().HasMaxLength(256)

			[Property](Function(t) t.JobTitle).IsRequired().HasMaxLength(50)

			[Property](Function(t) t.MaritalStatus).IsRequired().IsFixedLength().HasMaxLength(1)

			[Property](Function(t) t.Gender).IsRequired().IsFixedLength().HasMaxLength(1)

			' Table & Column Mappings
			ToTable("Employee", "HumanResources")
			[Property](Function(t) t.BusinessEntityID).HasColumnName("BusinessEntityID")
			[Property](Function(t) t.NationalIDNumber).HasColumnName("NationalIDNumber")
			[Property](Function(t) t.LoginID).HasColumnName("LoginID")
			[Property](Function(t) t.JobTitle).HasColumnName("JobTitle")
			[Property](Function(t) t.DateOfBirth).HasColumnName("BirthDate")
			[Property](Function(t) t.MaritalStatus).HasColumnName("MaritalStatus")
			[Property](Function(t) t.Gender).HasColumnName("Gender")
			[Property](Function(t) t.HireDate).HasColumnName("HireDate")
			[Property](Function(t) t.Salaried).HasColumnName("SalariedFlag")
			[Property](Function(t) t.VacationHours).HasColumnName("VacationHours")
			[Property](Function(t) t.SickLeaveHours).HasColumnName("SickLeaveHours")
			[Property](Function(t) t.Current).HasColumnName("CurrentFlag")
			[Property](Function(t) t.rowguid).HasColumnName("rowguid")
			[Property](Function(t) t.ModifiedDate).HasColumnName("ModifiedDate") '.IsConcurrencyToken();

			' Relationships
			Ignore(Function(t) t.Manager)
			HasOptional(Function(t) t.SalesPerson).WithRequired(Function(t) t.EmployeeDetails)
			HasRequired(Function(t) t.PersonDetails).WithOptional(Function(t) t.Employee)
		End Sub
	End Class

	Public Module Mapper
		<System.Runtime.CompilerServices.Extension> _
		Public Sub Map(ByVal builder As EntityTypeBuilder(Of Employee))
			builder.HasKey(Function(t) t.BusinessEntityID)

			' Properties
			builder.Property(Function(t) t.NationalIDNumber).IsRequired().HasMaxLength(15)

			builder.Property(Function(t) t.LoginID).IsRequired().HasMaxLength(256)

			builder.Property(Function(t) t.JobTitle).IsRequired().HasMaxLength(50)

			builder.Property(Function(t) t.MaritalStatus).IsRequired().IsFixedLength().HasMaxLength(1)

			builder.Property(Function(t) t.Gender).IsRequired().IsFixedLength().HasMaxLength(1)

			' Table & Column Mappings
			builder.ToTable("Employee", "HumanResources")
			builder.Property(Function(t) t.BusinessEntityID).HasColumnName("BusinessEntityID")
			builder.Property(Function(t) t.NationalIDNumber).HasColumnName("NationalIDNumber")
			builder.Property(Function(t) t.LoginID).HasColumnName("LoginID")
			builder.Property(Function(t) t.JobTitle).HasColumnName("JobTitle")
			builder.Property(Function(t) t.DateOfBirth).HasColumnName("BirthDate")
			builder.Property(Function(t) t.MaritalStatus).HasColumnName("MaritalStatus")
			builder.Property(Function(t) t.Gender).HasColumnName("Gender")
			builder.Property(Function(t) t.HireDate).HasColumnName("HireDate")
			builder.Property(Function(t) t.Salaried).HasColumnName("SalariedFlag")
			builder.Property(Function(t) t.VacationHours).HasColumnName("VacationHours")
			builder.Property(Function(t) t.SickLeaveHours).HasColumnName("SickLeaveHours")
			builder.Property(Function(t) t.Current).HasColumnName("CurrentFlag")
			builder.Property(Function(t) t.rowguid).HasColumnName("rowguid")
			builder.Property(Function(t) t.ModifiedDate).HasColumnName("ModifiedDate") '.IsConcurrencyToken();

			' Relationships
			builder.Ignore(Function(t) t.Manager)
			builder.HasOne(Function(t) t.PersonDetails).WithOne(Function(t) t.Employee)
		End Sub
	End Module
End Namespace